src/praisonai/praisonai/cli/features/persistence.py

```
@@ -562,20 +562,29 @@ def _mask_url(url: str) -> str:
+
+    Delegates to ``PraisonAIDB._detect_backend`` (the single, tested source of
+    truth) so the doctor recognises the same schemes as the live path
+    (e.g. ``libsql://`` -> turso, ``*.supabase.co`` -> supabase). Unknown
+    schemes fall back to the doctor's lenient ``default`` instead of failing.
+    """
+    try:
+        from praisonai.db.adapter import PraisonAIDB
+        return PraisonAIDB._detect_backend(PraisonAIDB, url)
+    except ValueError:
+        return default
+    except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)